spec/features/hq/crud_sponsor_spec.rb

```diff
@@ -0,0 +1,190 @@
+    fill_in 'Name', :with => 'Sponsor1'
+    select 'Active', :from => 'Status'
+    select 'Male', :from => 'Gender'
+    fill_in 'Start date', :with => '2014-08-25'
+    fill_in 'Requested orphan count', :with => '2'
+    select 'Individual', :from => 'sponsor_sponsor_type_id'
+    select Branch.all.first.name, :from => 'sponsor_branch_id'
+    select '(Afghanistan) أفغانستان', :from => 'Country'
+    select Sponsor.all_cities.first, :from => 'City'
+    select @sponsors.first.agent.user_name, :from => 'sponsor_agent_id'
+    click_button 'Create Sponsor'
+    and_i_should_be_on :hq_sponsor_path, { sponsor_name: 'Sponsor1' }
+    and_i_should_see 'Sponsor successfuly saved'
+    and_i_should_see 'Sponsor1'
+  end
+
+  scenario 'Should be able to edit a sponsor from the sponsor show page' do
+    name = @sponsors.first.name
+    given_i_am_on_page_for_sponsor 'Show Sponsor', name
+    click_link 'Edit Sponsor'
+    and_i_should_be_on :edit_hq_sponsor_path, { sponsor_name: name }
+    fill_in 'Additional info', :with => 'Additional Information'
+    click_button 'Update Sponsor'
+    and_i_should_be_on :hq_sponsor_path, { sponsor_name: name }
+    and_i_should_see 'Sponsor successfuly saved'
+    and_i_should_see 'Additional Information'
+  end
+
+  scenario "can't edit sponsor_type, organization or branch" do
+    given_i_am_on_page_for_sponsor 'Edit Sponsor', @sponsors.first.name
+    i_should_not_be_able_to_change 'Sponsor type', 'sponsor'
+    i_should_not_be_able_to_change 'Organization', 'sponsor'
+    i_should_not_be_able_to_change 'Branch', 'sponsor'
+  end
+
+  scenario "can't delete a sponsor from the sponsor show page" do
+    given_i_am_on_page_for_sponsor 'Show Sponsor', @sponsors.first.name
+    and_i_should_not_see 'Delete Sponsor'
+  end
+
+  scenario "returns to partner show page on cancelling edit partner" do
+    name = @sponsors.first.name
+    given_i_am_on_page_for_sponsor 'Show Sponsor', name
+    click_link 'Edit Sponsor'
+    fill_in 'Additional info', :with => 'Additional Information'
+    click_link 'Cancel'
+    and_i_should_be_on :hq_sponsor_path, { sponsor_name: name }
+    and_i_should_not_see 'Additional Information'
+  end
+
+  private
+
+  def i_should_see_the_following_fields_on_the_page table
+    table.each do |hash|
+      expect(page).to have_content(hash[:value])
+    end
+  end
+
+  def given_i_am_on_page_for_sponsor page_name, sponsor_name = nil
+    if sponsor_name.nil?
+      visit path_to_sponsor_role page_name
+    else
+      sponsor = Sponsor.find_by name: sponsor_name unless sponsor_name.nil?
+      visit path_to_sponsor_role page_name, sponsor.id
+    end
+  end
+
+  def path_to_sponsor_role page_name, id = nil
+    name = page_name.downcase
+    case name
+      when 'edit sponsor' then
+        edit_hq_sponsor_path id
+      when 'show sponsor' then
+        hq_sponsor_path id
+      when 'new sponsor' then
+        new_hq_sponsor_path
+    end
+  end
```

Comment:
I have pushed all the other changes suggested. However, this change probably needs further thought ...

I agree in principle but how do you plan on approaching this?
- `path_to_sponsor_role` could be incorporated into `get_path` if `get_path` had no expectations
- unless you rename `and_i_should_be_on` the method name gives the impression the method has an expectation
- If you remove the expectations and make them helpers, given that these expectations are needed, otherwise they wouldn't be there, where do you put them? Are you wanting to push them back to the scenario in question. If so how much does this break dry?
- `path_to` has a mixture of paths ending in path or page, perhaps we should use just one for consistency.
- `and_i_should_be_on` seems to require a single key hash to work, I can't put my finger on it but it doesn't sit easy with me. It seems a little brittle.

Regardless, it is probably worth refactoring before things get even more complicated.

Thoughts?